datalad/__init__.py

```diff
@@ -142,6 +142,19 @@ def set_envvar(v, val):
 """
     from datalad.support.external_versions import external_versions
     if external_versions['cmd:git'] < "2.32":
+        if on_osx or on_windows:
+            # This approach of changing HOME (or USERPROFILE respectively)
+            # has unintended side-effects on:
+            #   1. Windows, generally, as any subprocess will fail to retrieve
+            #      correct standard locations for configs, caches, etc.
+            #   2. OSX, most noteably the system-level osxkeychain
+            #      git-credential helper, which will lead to hanging git
+            #      commands if it fails to find the keychain in its expected
+            #      location under $HOME/Library/Keychain.
+            raise RuntimeError("Refuse to execute datalad tests with git < 2.32 "
+                               "on this system, as this may have unintended "
+                               "side-effects.")
+
```

Comment:
Without the knowledge acquired over the course of #6160 and #6260, this error message is pretty confusing. It would be fair to add a link or reference. Same goes for the comment above.

Reply:
Agreed